src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -134,5 +127,329 @@ private BoundExpression BindInterpolatedString(InterpolatedStringExpressionSynta
+            // type that makes it to this method is actually an interpolated string builder type, and we should fully report any binding errors
+            // we encounter while doing this work.
+
+            if (source.Parts.IsEmpty)
+            {
+                return (ImmutableArray<BoundExpression>.Empty, false);
+            }
+
+            var implicitBuilderReceiver = new BoundInterpolatedStringBuilderPlaceholder(source.Syntax, builderType);
+            bool? builderPatternExpectsBool = null;
+            var builderAppendCalls = ArrayBuilder<BoundExpression>.GetInstance(source.Parts.Length);
+            var argumentsBuilder = ArrayBuilder<BoundExpression>.GetInstance(3);
+            var parameterNamesAndLocationsBuilder = ArrayBuilder<(string, Location)?>.GetInstance(3);
+
+            foreach (var part in source.Parts)
+            {
+                Debug.Assert(part is BoundLiteral or BoundStringInsert);
+                argumentsBuilder.Clear();
+                parameterNamesAndLocationsBuilder.Clear();
+                string methodName;
+
+                if (part is BoundStringInsert insert)
+                {
+                    methodName = "AppendFormatted";
+                    argumentsBuilder.Add(insert.Value);
+                    parameterNamesAndLocationsBuilder.Add(null);
+
+                    if (insert.Alignment is not null)
+                    {
+                        argumentsBuilder.Add(insert.Alignment);
+                        parameterNamesAndLocationsBuilder.Add(("alignment", insert.Alignment.Syntax.Location));
+                    }
+                    if (insert.Format is not null)
+                    {
+                        argumentsBuilder.Add(insert.Format);
+                        parameterNamesAndLocationsBuilder.Add(("format", insert.Format.Syntax.Location));
+                    }
+                }
+                else
+                {
+                    methodName = "AppendLiteral";
+                    argumentsBuilder.Add(part);
+                }
+
+                var arguments = argumentsBuilder.ToImmutableAndClear();
+                ImmutableArray<(string, Location)?> parameterNamesAndLocations;
+                if (parameterNamesAndLocationsBuilder.Count > 1)
+                {
+                    parameterNamesAndLocations = parameterNamesAndLocationsBuilder.ToImmutableAndClear();
+                }
+                else
+                {
+                    Debug.Assert(parameterNamesAndLocationsBuilder.Count == 0 || parameterNamesAndLocationsBuilder[0] == null);
+                    parameterNamesAndLocations = default;
+                    parameterNamesAndLocationsBuilder.Clear();
+                }
+
+                var call = MakeInvocationExpression(part.Syntax, implicitBuilderReceiver, methodName, arguments, diagnostics, names: parameterNamesAndLocations, searchExtensionMethodsIfNecessary: false);
+                builderAppendCalls.Add(call);
```

Comment:
>call [](http://example.com/codeflow?start=39&length=4)

Is there a guarantee that the node is marked as compiler generated? Otherwise we should consider marking it here.

Reply:
There is. `MakeInvocationExpression` is specifically for compiler-synthesized calls and unconditionally marks the invocation it returns as compiler-generated.

Reply:
> There is. MakeInvocationExpression is specifically for compiler-synthesized calls and unconditionally marks the invocation it returns as compiler-generated.

It looks like there is a an early return from that method that doesn't do that.

Reply:
That does feel more like a bug in MakeInvocationExpression, but alright, I can adjust.